### ISSUE TITLE
OSError: cpp_functions.so: undefined symbol: gzopen

### ISSUE THREAD

**Issue by tomasyeo** [rrwick/Unicycler#18]
hi Ryan,
I have successfully installed Unicycler. But I get the following run error:
```
Traceback (most recent call last):
  File "./unicycler-runner.py", line 18, in <module>
    from unicycler.unicycler import main
  File "/opt/NGS_tools/Unicycler/unicycler/unicycler.py", line 25, in <module>
    from .assembly_graph import AssemblyGraph
  File "/opt/NGS_tools/Unicycler/unicycler/assembly_graph.py", line 18, in <module>
    from .assembly_graph_segment import Segment
  File "/opt/NGS_tools/Unicycler/unicycler/assembly_graph_segment.py", line 19, in <module>
    from .bridge_long_read import LongReadBridge
  File "/opt/NGS_tools/Unicycler/unicycler/bridge_long_read.py", line 25, in <module>
    from .cpp_wrappers import consensus_alignment
  File "/opt/NGS_tools/Unicycler/unicycler/cpp_wrappers.py", line 27, in <module>
    C_LIB = CDLL(SO_FILE_FULL)
  File "/opt/anaconda3/lib/python3.5/ctypes/__init__.py", line 347, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /opt/NGS_tools/Unicycler/unicycler/cpp_functions.so: undefined symbol: gzopen
```
I'm using Ubuntu 16.04 with g++ (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0.  
```
zlib1g/xenial-updates,now 1:1.2.8.dfsg-2ubuntu4.1 amd64 [installed]
  compression library - runtime
zlib1g-dev/xenial-updates,now 1:1.2.8.dfsg-2ubuntu4.1 amd64 [installed,automatic]
  compression library - development
```
Is there a problem with my dependencies? thanks.

regards,
tomas

**Comment by andersgs** [comment]
Hi Tomas.

It looks like you are using Conda. How did you install `unicycler`? When installing on our system, which uses Brew, I have to `pip install --no-binaries :all:` to force a local compilation rather than use the pre-compiled binaries. That is because standard libraries (like libz) are not kept at default Linux locations (e.g.,, `/usr/lib`, or `/usr/local/lib`). Try uninstalling `unicycler` and installing with the `--no-binaries :all:` option. It will take a little longer, but hopefully will produce binaries that are locally configured to your system.

Good luck.

Anders.

**Comment by tomasyeo** [comment]
Hi Anders,
Thanks for the tip. It didn't work and there is nothing wrong with the zlib1g libraries on ubuntu 16.04. 

The problem lies in the **Makefile**, line 105:
```
$(CXX) $(FLAGS) $(CXXFLAGS) $(LDFLAGS) -Wl,$(SONAME),$(TARGET) -o $(TARGET) $(OBJECTS)
```

So I positioned _LDFLAGS_ after _-o_ in the following:
```
$(CXX) $(FLAGS) $(CXXFLAGS) -Wl,$(SONAME),$(TARGET) -o $(TARGET) $(OBJECTS) $(LDFLAGS)
```
And unicycler works without the offending _undefined symbol: gzopen_ error.

regards,
tomas

**Comment by andersgs** [comment]
Good sleuthing @tomasyeo. Out of interest, what compiler and version are you using?

**Comment by tomasyeo** [comment]
@andersgs 
its g++ (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0.

**Comment by swlong** [comment]
I've run into this same error using a Biolinux VM. Ubuntu 14.04, running g++ 4.9.1.

Fixed it by altering the Makefile as described above.

**Comment by rrwick** [comment]
Thank you, everybody! I'm no expert with Makefiles, so I'm not surprised something in there wasn't quite right.

I just put this change into the development branch. It should hopefully make its way to a new release soon. Since I just did a big feature release (v0.4.0) I'm hoping to make a bug-fix release (v0.4.1) in the not-too-distance future.
